src/db.c

```diff
@@ -834,8 +834,13 @@ void flushallSyncBgDone(uint64_t client_id, void *sflush) {
     /* mark client as unblocked */
     unblockClient(c, 1);
 
-    /* FLUSH command is finished. resetClient() and update replication offset. */
-    commandProcessed(c);
+    if (c->flags & CLIENT_PENDING_COMMAND) {
```

Comment:
this, combined with the check in unblockClient, would mean that resetClient will be performed only once, right?
but maybe the call to `resetClient` and `reqresAppendResponse` should be converted to a call for commandProcessed?
on one hand, the command was aborted in that flow, but on the other hand, it was replied, so in that way it was processed.

Reply:
Do you mean change the resetClient and reqresAppendResponse calls in unblock to call commandProcessed()?
This is a bit strange, if there is no `CLIENT_PENDING_COMMAND` flag, we will ignore the current execution command.
If calling commandProcessed in unblock is a bit confusing, we don't actually reprocess this command, we don't even do any cleaning up.
Another point is that `commandProcessed()` is prohibited from being called by a blocked client, and we will have to move `if (!(c->flags & CLIENT_PENDING_COMMAND) && c->bstate.btype != BLOCKED_SHUTDOWN)` down, not sure if there is a risk.

Reply:
ok, let's drop it.
i didn't look too much into it. just saw two places doing resetClient + reqresAppendResponse, and thought it's dangerous since we can add / change one, and forget the other.